upup/pkg/fi/cloudup/apply_cluster.go

```diff
@@ -230,7 +230,9 @@ func (c *ApplyClusterCmd) Run() error {
 
 				// Autoscaling
 				"autoscalingGroup":    &awstasks.AutoscalingGroup{},
-				"launchConfiguration": &awstasks.LaunchConfiguration{},
+				"launchConfiguration": &awstasks.LaunchConfiguration{
+					AssociatePublicIP: c.Cluster.Spec.AssociatePublicIP,
```

Comment:
Ah ... this isn't how it was intended to be used.  Instead, you should use a template helper function, which you define here:

https://github.com/kubernetes/kops/blob/master/upup/pkg/fi/cloudup/template_functions.go#L55

Then if it's a cluster spec that's pretty easy, because there's only one of them.  If it is on the InstanceGroup spec though, you can just pass the current instance group in as a variable like this example:

https://github.com/kubernetes/kops/blob/master/upup/models/cloudup/_aws/master/_master_single/master_single.yaml#L41